redis/sentinel.py

```diff
@@ -123,18 +120,42 @@ def rotate_slaves(self):
             yield self.get_master_address()
         except MasterNotFoundError:
             pass
-        raise SlaveNotFoundError('No slave found for %r' % (self.service_name))
-
-    def _checkpid(self):
-        if self.pid != os.getpid():
-            self.disconnect()
-            self.reset()
-            self.__init__(self.service_name, self.sentinel_manager,
-                          is_master=self.is_master,
-                          check_connection=self.check_connection,
-                          connection_class=self.connection_class,
-                          max_connections=self.max_connections,
-                          **self.connection_kwargs)
+        raise SlaveNotFoundError('No slave found for %r' % (self.service_name,))
+
+    def _check_connection(self, connection):
+        if connection.to_be_disconnected:
+            connection.disconnect()
+            self.get_master_address()
+            return False
+        if self.is_master:
+            if self.master_address != (connection.host, connection.port):
```

Comment:
Shouldn't a similar check be made for the case when `self.is_master` is `False`? (i.e. you called `slave_for` before and the slave you are connected to changed to `master`)

Reply:
`slave_for` can actually also give you a master connection (only as a fallback, but it still can), so I consider `slave_for` as "give me a connection to any redis server, preferably slave" and don't think we need to disconnect such connections

Reply:
Ah, okay then.